Implement basic integration test

Toshbrown opened the pull request:
Addresses issue https://github.com/me-box/databox/issues/55 for the databox main repo

This test set of test will:

1. Build all the core components 
2. Strat databox
3. Check the container manager UI is available 
4. Check the arbiter is available 
5. Install driver-os-monitor 
6. Install app-os-monitor 
7. Check that the app-os-monitor UI is available

Toshbrown committed:
Eanble 'is it up test'

Toshbrown committed:
Eanble 'is it up test'

Toshbrown committed:
Don't stream log output in when testing

Toshbrown committed:
Don't stream log output in when testing

Toshbrown committed:
Test if the arbitor is working

Toshbrown committed:
Test if the arbitor is working

Toshbrown committed:
Test Driver and App install 

uses driver-os-monitor and app-os-monitor to check basic functionality

Toshbrown committed:
Test Driver and App install 

uses driver-os-monitor and app-os-monitor to check basic functionality

Toshbrown committed:
Build platform-app-store in dev mode

Toshbrown commented:
Test fails due to missing platform-app-store fix incoming

Toshbrown committed:
Fix build section of yaml file

Toshbrown committed:
Merge pull request #96 from me-box/fix/missing-appserver

Fix/missing appserver

Toshbrown committed:
Merge remote-tracking branch 'origin/fet/travis' into fet/travis

Toshbrown committed:
Only build want we need in testing

Should speed things up abit

Toshbrown committed:
Make test fail!!!

And output debug info

Toshbrown committed:
Wait a bit longer

travis can take a while to start up databox

Toshbrown committed:
Remove uneeded logging

Toshbrown committed:
Wait longer jic

But i think its a docker version issue

Toshbrown committed:
Sleeping for longer seems to help

lets seep evan more

Toshbrown committed:
Debuging services not starting on travis

Toshbrown committed:
Looks like builds are failing 

lets try prebuild images just to make sure

Toshbrown committed:
Remove store-timeseries

Untill the build is faster ;-)

Toshbrown committed:
Update docker

Toshbrown committed:
Fix app-store build

Toshbrown committed:
change wait times

Toshbrown committed:
Move from using docker compose for building

With docker compose its hard to tell if one of the many builds fail. Making it harder to track down errors. Moving to a build script gives us more control

Toshbrown committed:
Fix image name

Toshbrown committed:
Debug output on exit

Toshbrown committed:
Fix built image names 

Face hits palm

Toshbrown committed:
Rename test script

Toshbrown committed:
Fix cmd not found

Toshbrown committed:
Stop if there is an error seeding the manifests

Toshbrown committed:
Always check for erros

Toshbrown committed:
Fix output

Toshbrown committed:
Ouput docker logs on build fail

mor1 commented:
Looks basically in the right direction.
Some typos in messages etc :)
Do you want me to try a full review on this yet, or are you still finishing the PR? (In which case I'll wait to review until it's ready I think.)

Toshbrown committed:
Typos

mor1 commented on .travis.yml:
```diff
@@ -1,7 +1,11 @@
 script:
-  - exit 0
-
+  - ./databox-test
```
put this in a `scripts` subdirectory or similar?

Toshbrown replied:
I quite like this here, might mean people get used to running it 👍

mor1 commented on databox-build-core:
```diff
@@ -0,0 +1,41 @@
+
+RED='\033[0;31m'
+GREEN='\033[0;32m' 
+NC='\033[0m'
```
could maybe wrap these up as `green`, `red` shell functions instead, to avoid need to remember to close the escape

Toshbrown replied:
done

mor1 commented on databox-build-core:
```diff
@@ -0,0 +1,41 @@
+
+build() 
+{
+    echo "[$(date +'%Y-%m-%dT%H:%M:%S%z') $ME]: Starting build ${1} ..."
```
Too many repeats of that very long `date ...` invocation. Refactor to shell functions that format the line appropriately?

Toshbrown replied:
done

mor1 commented on databox-start:
```diff
@@ -16,6 +16,12 @@ die() {
   exit $rc
 }
+assert() {
+  if [ "$1" != "$2" ] 
```
Google shell script style guide suggests `[[ ... ]]` is better
(https://google.github.io/styleguide/shell.xml)

Toshbrown replied:
done

mor1 commented on databox-test:
```diff
@@ -0,0 +1,75 @@
+
+RED='\033[0;31m'
+GREEN='\033[0;32m' 
+NC='\033[0m'
```
definitely factor these out into some utility library to be `source`d I think

Toshbrown replied:
done

mor1 commented on databox-test:
```diff
@@ -0,0 +1,75 @@
+docker ps
+
+#can we see the CM UI
+STATUS=$(curl -sL -w "%{http_code}\\n" "http://127.0.0.1:8989/" -o /dev/null)
```
odd use of `-o /dev/null`? i would usually just redirect to `/dev/null` and not use the `-o` flag. is there a reason?

Toshbrown replied:
because 2>&1 /dev/null does not work.

mor1 replied:
Ah. `-w` causes display to `stdout`. Which helpfully causes it to intermingle with the fetched content. Ok. Less of a hack than piping through `tail` I guess.

mor1 commented on docker-compose-dev-local-images.yaml:
```diff
@@ -3,24 +3,12 @@ version: '3'
 
 # Build only (put your images in here)
 services:
```
why are the stanzas below being removed from the dev version?

Toshbrown replied:
store-json, driver-os-monitor and app-os-monitor are needed for testing so have been moved into core as they are no longer optional.   Store-json will be replaced by store-timeseries when its ready.

mor1 commented on docker-compose.yaml:
```diff
@@ -1,13 +1,8 @@
-
 services:
-#Build and satart the databox platform
+#satart the databox platform
```
typo (satart)

Toshbrown replied:
done

mor1 commented on docker-compose.yaml:
```diff
@@ -79,7 +68,7 @@ services:
                 - 'node.role == manager'
 
-
+        
```
junk whitespace

Toshbrown replied:
done

Toshbrown committed:
Create scripts utilities 

Has 

red() - makes $1 red 
green() - makes $1 green 
datef() - prints formated date

Toshbrown committed:
Use scripts/utils

Toshbrown committed:
Stop the app store

Toshbrown committed:
Fix typo and junk whitespace
